packages/controllers/file_upload.coffee

```diff
@@ -0,0 +1,19 @@
+Template.fileUpload.onCreated ->
+  @fileName = new ReactiveVar('No file selected')
+  @hasFile = new ReactiveVar(false)
+
+Template.fileUpload.helpers
+  fileName: ->
+    Template.instance().fileName.get()
+  hasFile: ->
+    if Template.instance().hasFile.get()
+      'has-file'
+
+Template.fileUpload.events
+  'change .btn-file': (event, template) ->
+    template.fileName.set(event.target.value.replace('C:\\fakepath\\',''))
```

Comment:
*According to the specifications of HTML5, a file upload control should not reveal the real local path to the file you have selected, if you manipulate its value string with JavaScript. Instead, the string that is returned by the script, which handles the file information is c:fakepath.*
http://martinivanov.net/2009/06/09/the-mystery-of-cfakepath-unveiled/